Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,242 @@
+1. `objectReference` to the generated `Secret`.
+1. phases of bucket creation:
+    - _Pending_: the operator is processing the request
+    - _Bound_: the operator finished processing the request and linked the OBC and OB
+    - _Released_: the OB has been deleted, leaving the OBC unclaimed but unavailable.
+    - _Failed_: not currently set.
+
+##### Generated Secret (sample for rook-ceph provider)
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ceph-bucket
+  namespace: rook-ceph
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  ownerReferences:
+  - name: ceph-bucket [1]
+    ...
+type: Opaque
+data:
+  ACCESS_KEY_ID: BASE64_ENCODED-1
+  SECRET_ACCESS_KEY: BASE64_ENCODED-2
+  ... [2]
+```
+1. ownerReference makes this secret a child of the originating OBC for clean up purposes.
+1. ACCESS_KEY_ID and SECRET_ACCESS_KEY are the only secret keys defined by the library.
+Provisioners are able to cause the lib to create additional keys by returning  the `AdditionalSecretConfig` field.
+**Note:** the library will create the Secret using `stringData:` and let the Secret API base64 encode the values.
+Eg:
+```
+stringData:
+  endpoint: |-
+    ACCESS_KEY_ID: NON-BASE64-STRING
+    SECRET_ACCESS_KEY: NON-BASE64-STRING
+```
+
+##### Generated ConfigMap (sample for rook-ceph provider)
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ceph-bucket
+  namespace: rook-ceph
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+  ownerReferences:
+  - name: rook-ceph
+    ...
+data:
+  BUCKET_HOST: http://my-store [1]
+  BUCKET_PORT: 80 [2]
+  BUCKET_NAME: photo-booth-62PrQ [3]
+  BUCKET_REGION: us-west-1
+  ... [4]
+```
+1. host URL.
+1. host port.
+1. unique bucket name.
+1. the above data keys are defined by the library.
+Provisioners are able to cause the lib to create additional data keys by returning the `AdditionalConfigData` field.
+
+##### App Pod (independent of provisioner)
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app-pod
+  namespace: dev-user
+spec:
+  containers:
+  - name: mycontainer
+    image: redis
+    envFrom: [1]
+    - configMapRef:
+        name: ceph-bucket [2]
+    - secretRef:
+        name: ceph-bucket [3]
+```
+1. use `env:` if mapping of the defined key names to the env var names used by the app is needed.
+1. makes available to the pod as env variables: `BUCKET_HOST`, `BUCKET_PORT`, `BUCKET_NAME`
+1. makes available to the pod as env variables: `ACCESS_KEY_ID`, `SECRET_ACCESS_KEY`
+
+##### Generated OB Custom Resource
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucket
+metadata:
+  name: obc-default-ceph-bucket
+  finalizers:
+  - objectbucket.io/finalizer
+  labels:
+    bucket-provisioner: aws-s3.io-bucket
+spec:
+  objectBucketSource: [1]
+    provider: ceph.rook.io/object
+  storageClassName: rook-ceph-bucket
+  claimRef: objectreference [2]
```

Comment:
Other uses of _objectReference_ in this doc are formatted as `objectReference`. I vote for consistency.